pkg/bisect/bisect.go

```diff
@@ -592,26 +592,29 @@ func (env *env) test() (*testResult, error) {
 	bad, good, infra, rep := env.processResults(current, results)
 	res.rep = rep
 	res.verdict = vcs.BisectSkip
-	if infra > len(results)/2 {
-		// More than 1/2 of runs failed with infrastructure error,
+	if infra*3 > len(results) {
+		// More than 1/3 of runs failed with an infrastructure error,
 		// no sense in continuing the bisection at the moment.
-		return res, &InfraError{Title: "more than 50% of runs failed with an infra error"}
-	}
-	if bad != 0 {
+		return res, &InfraError{Title: "more than 1/3 of runs failed with an infra error"}
+	}
+	if (good+bad)*3 < (len(results)-infra)*2 {
```

Comment:
why *2?

Reply:
I had to do some math that I cannot do in my head to figure out that we require at least 5 non-failed results if infra errors happened for non-flaky case, 7 if no infra error happened, 9 for flaky with infra errors, and 14 without infra errors.
And I am still not sure if I did the math correctly.

1. Can we make the logic simpler so that I can interpret it in my head?
2. Why does the number depend on presence of infra errors? 9 vs 14 for flaky case looks esp problematic. We concluded that 10 is not enough for reliable decision, and now we use only 9.